File: android/app/src/main/java/de/hastnetwork/jarvis/data/model/Agent.kt
package de.hastnetwork.jarvis.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Mirrors the `Agent` entity from architecture.md §4.
 */
@Serializable
data class Agent(
    val id: String,
    val name: String,
    val description: String,
    @SerialName("system_prompt") val systemPrompt: String,
    @SerialName("ollama_model") val ollamaModel: String,
    @SerialName("voice_id") val voiceId: String,
    @SerialName("avatar_color") val avatarColor: String,
    @SerialName("is_default") val isDefault: Boolean = false,
)

/** Body for `POST /api/agents` — same shape as [Agent] but without `id`. */
@Serializable
data class AgentCreateRequest(
    val name: String,
    val description: String,
    @SerialName("system_prompt") val systemPrompt: String,
    @SerialName("ollama_model") val ollamaModel: String,
    @SerialName("voice_id") val voiceId: String,
    @SerialName("avatar_color") val avatarColor: String,
    @SerialName("is_default") val isDefault: Boolean = false,
)

File: backend/app/models/agent.py
from sqlmodel import Field, SQLModel


class AgentBase(SQLModel):
    name: str
    description: str = ""
    system_prompt: str
    ollama_model: str
    voice_id: str = "default"
    avatar_color: str = "#4285F4"
    is_default: bool = False


class Agent(AgentBase, table=True):
    id: str = Field(primary_key=True)


class AgentCreate(AgentBase):
    id: str


class AgentUpdate(SQLModel):
    name: str | None = None
    description: str | None = None
    system_prompt: str | None = None
    ollama_model: str | None = None
    voice_id: str | None = None
    avatar_color: str | None = None
    is_default: bool | None = None


class AgentRead(AgentBase):
    id: str
